rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,106 @@
+---
+authors: Andrew LeFevre (andrew.lefevere@goteleport.com)
+state: draft
+---
+
+# RFD 0194 - Official SELinux Policy for Teleport SSH Agents
+
+## Required Approvers
+
+- Engineering: (@r0mant && @rosstimothy)
+- Security: Doyensec
+
+## What
+
+This RFD proposes an official SELinux policy for Teleport SSH agents. The policy will enforce that minimal permissions are given to SSH agents,
+and that security boundaries between the main server process and child session processes are respected.
+
+The policy will be bundled in Teleport releases, and be updated and managed by the `teleport-update` agent updater (see https://github.com/gravitational/teleport/blob/master/rfd/0184-agent-auto-updates.md).
+
+## Why
+
+Teleport is often deployed in environments with strict security requirements. SELinux provides a way to enforce fine-grained access controls, which
+can help prevent privilege escalation and lateral movement, provide peace of mind, and comply with security standards. As agents are the most widely
+deployed component of Teleport, it is important to ensure that they are secure and compliant.
+
+Only SSH agents are initially in scope for this proposal as a starting point, but the SELinux policy may be extended to other components of Teleport in the future.
+
+## Details
+
+### SELinux Policy Structure
+
+SSH agents need a lot of privileges to function properly. They need to bind to privileged ports, manage cgroups, create processes, and more.
+When an SSH session is started, the SSH service will create 2 child processes to handle it: 
+
+- An intermediary process that prepares another child process that will actually run the SSH session. Depending on what is configured, PAM contexts
+may be created, users may be created, a shell will be created, the user and group will be changed if necessary, and so on.
+- The intermediary process will then create the child session process, which only needs standard permissions of the host user that the SSH session
+was created for.
+
+Additionally, the SSH agent may create a networking child process to handle X11 forwarding, port forwarding, and SSH Agent forwarding. This process is a child process of the intermediary process as well.
+
+4 main SELinux domains will be created to ensure each process in the chain is granted only the necessary permissions: `teleport_ssh_agent`, `teleport_ssh_agent_intermediary`, `teleport_ssh_agent_session`, and `teleport_ssh_networking` respectively.
+
+Configurable file and port contexts will be created for the `teleport_ssh_agent` domain, and configurable file contexts
+will be created for the `teleport_ssh_agent_intermediary` domain to allow those processes to bind to the necessary ports and read and write
+to the necessary files, depending on the SSH agent configuration.
+
+SELinux booleans will also be created that will allow selective enabling or disabling of groups of permissions, depending on what is enabled
+in the Teleport agent configuration. 
+
+These include:
```

Comment:
Do we also need to consider a policy for the service to read from it's own data directory, including the audit logs saved to disk before being emitted to the auth server? Do we need a policy for dialing, e.g. to connect to the auth server to send the audit logs (or other integrations the node may have with other services)?

It would be worthwhile to add a section (or link to some docs) about what types of actions exactly require an SELinux policy, so that it will get the proper consideration in future features.